### PR TITLE
chore: kms encryption for admin api key

### PR DESCRIPTION
**Pull request by dracarys18**

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -66,8 +66,12 @@ echo "$(tput bold)$(tput setaf 1)If you need Card Vault, please create master ke
 echo "$(tput bold)$(tput setaf 3)To generate the master key, you can use the utility bundled within \n(https://github.com/juspay/hyperswitch-card-vault)$(tput sgr0)"
 echo "$(tput bold)$(tput setaf 3)If you have cargo installed you can run \n(cargo install --git https://github.com/juspay/hyperswitch-card-vault --bin utils --root . && ./bin/utils master-key && rm ./bin/utils && rmdir ./bin)$(tput sgr0)"
 
+echo "Please input the AES256 compatible master encryption key"
+read -s MASTER_ENC_KEY
+
```
Comment (nitesh-balla) [inline install.sh:71]: Let us update one click deployment files `production.yml` and `single-click.sh` files for the same.

File: install.sh [hunk 2]
```diff
@@ -92,7 +96,10 @@ if cdk deploy --require-approval never -c db_pass=$DB_PASS -c admin_api_key=$ADM
   CONTROL_CENTER_HOST=$(kubectl get ingress hyperswitch-control-center-alb-ingress -n hyperswitch -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
   SDK_HOST=$(kubectl get ingress hyperswitch-sdk-alb-ingress -n hyperswitch -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
   SDK_URL=$(aws cloudformation describe-stacks --stack-name hyperswitch --query "Stacks[0].Outputs[?OutputKey=='HyperLoaderUrl'].OutputValue" --output text)
-  # Deploy the hyperswitch application with the load balancer host name
+  echo "Please input the encrypted master key (optional): "
+  read -s MASTER_KEY
```
Comment (nitesh-balla) [inline install.sh:100]: Where are we using this `MASTER_KEY`?

File: lib/aws/encryption.py
Comment (NishantJoshi00) [inline lib/aws/encryption.py:1]: can you instead update the kms lambda that is written for card vault to be general purpose?

Comment (dracarys18) [reply lib/aws/encryption.py:1]: I will create an issue for this we can pick this during CDK rewrite?

**Comment by dracarys18** [comment]
All hyperswitch pods are running

Screenshot 2024-01-12 at 5 33 24 PM

CDK script successfull

Screenshot 2024-01-12 at 5 33 59 PM

File: lib/aws/eks.ts [hunk 3]
```diff
@@ -250,20 +390,28 @@ export class EksStack {
             },
           },
         },
+        postgresql: {
```
Comment (jagan-jaya) [inline lib/aws/eks.ts:393]: use existing helm values

File: lib/aws/eks.ts [hunk 4]
```diff
@@ -250,20 +390,28 @@ export class EksStack {
...
+                        database: "hyperswitch",
+                        password: kmsSecrets.kms_encrypted_db_pass,
+                    },
+                },
+            },
+        },
         loadBalancer: {
           targetSecurityGroup: lbSecurityGroup.securityGroupId,
         },
         redis: {
```
Comment (jagan-jaya) [inline lib/aws/eks.ts:409]: use existing helm values

**Review by jagan-jaya — CHANGES_REQUESTED** [review]

**Comment by nitesh-balla** [comment]
@dracarys18 Please resolve the conflicts. 
other than that LGTM!

**Comment by dracarys18** [comment]
Done

**Review by nitesh-balla — APPROVED** [review]

**Pull request merged**